syntax = "proto3";

package messenger;

message ResultReply {
	enum Result {
		SUCCESS = 0;
		FAILED = 1;
	}
	
	Result result = 1;
}

message LoginRequest {
	string user_name = 1;
}

message LoginReply {
	ResultReply result = 1;
	uint32 user_id = 2;
	uint64 time_stamp = 3;
}

message TextMessage {
	uint32 user_id = 1;
	string text_message = 2;
	uint64 time_stamp = 3;
}

message PendingMessagesRequest {
	uint64 time_stamp = 1;
}

message PendingMessagesReply {
	ResultReply result = 1;
	repeated string text_messages = 2;
}

service Messenger {
	rpc Login (LoginRequest) returns (LoginReply) {}
	rpc SendTextMessage (TextMessage) returns (ResultReply) {}
	rpc RequestPendingMessages (PendingMessagesRequest) returns (PendingMessagesReply) {}
}